src/UglyToad.PdfPig/PdfFonts/Parser/Handlers/Type3FontHandler.cs

```diff
@@ -49,11 +49,28 @@ public IFont Generate(DictionaryToken dictionary)
                     toUnicodeCMap = CMapCache.Parse(new ByteArrayInputBytes(decodedUnicodeCMap));
                 }
             }
-            
-            return new Type3Font(NameToken.Type3, boundingBox, fontMatrix, encoding, firstCharacter,
+
+            var name = GetFontName(dictionary);
+
+            return new Type3Font(name, boundingBox, fontMatrix, encoding, firstCharacter,
                 lastCharacter, widths, toUnicodeCMap);
         }
 
+        private NameToken GetFontName(DictionaryToken dictionary)
+        {
+            if (!dictionary.TryGet(NameToken.Name, out var fontName))
```

Comment:
This might be better as:
```suggestion
            if (!dictionary.TryGet(NameToken.Name, scanner, out NameToken fontName))
```

If I remember correctly the overload which takes a `IPdfTokenScanner` which should be available here will correctly find the name if it lives in a different object, which sometimes happens, otherwise the change looks good to me :+1: